Add accessors to pathogen genomes

ttung opened the pull request:
### Description
1. Add accessor to retrieve all the pathogen genomes retrieved from a HostFilteredSequencingReadsCollection.
2. Add accessor to retrieve all the pathogen genomes retrieved from a SequencingReadsCollection.

#### Issue
[ch126782](https://app.clubhouse.io/genepi/story/126782/workflow-to-pull-the-data-from-aspen-api-endpoint-and-launch-nextstrain-build)

### Test plan
used in subsequent PR.

phoenixAja commented on src/py/aspen/database/models/host_filtering.py:
```diff
@@ -23,14 +26,26 @@ class HostFilteredSequencingReadsCollection(Entity):
 
+    @property
+    def pathogen_genomes(self) -> Sequence[CalledPathogenGenome]:
+        # this import has to be here for circular dependencies reasons. :(
+        from aspen.database.models.sequences import CalledPathogenGenome
```
i feel like we should reorganize this so that we don't have to do this (if that's possible)

phoenixAja commented on src/py/aspen/database/models/host_filtering.py:
```diff
@@ -23,14 +26,26 @@ class HostFilteredSequencingReadsCollection(Entity):
         from aspen.database.models.sequences import SequencingReadsCollection
 
         parents = self.get_parents(SequencingReadsCollection)
         assert len(parents) == 1
```
if this were to fail would that mean that it was due to a data integrity issue?

ttung replied:
It would imply that a host filtering workflow was created that had more than one `SequencingReadsCollection` input.

Commit:
Add accessors to pathogen genomes

### Description
1. Add accessor to retrieve all the pathogen genomes retrieved from a HostFilteredSequencingReadsCollection.
2. Add accessor to retrieve all the pathogen genomes retrieved from a SequencingReadsCollection.

#### Issue
[ch126782](https://app.clubhouse.io/genepi/story/126782/workflow-to-pull-the-data-from-aspen-api-endpoint-and-launch-nextstrain-build)

### Test plan
used in subsequent PR.
